pkgs/applications/science/electronics/vivado/default.nix

```
@@ -0,0 +1,127 @@
+{ stdenv
+, lib
+, coreutils
+, fetchurl
+, patchelf
+, procps
+, makeWrapper
+, requireFile
+, ncurses5
+, zlib
+, libuuid
+, libSM
+, libICE
+, libX11
+, libXrender
+, libxcb
+, libXext
+, libXtst
+, libXi
+, glib
+, freetype
+, gtk2
+}:
+
+stdenv.mkDerivation rec {
+  pname = "vivado";
+  version = "2017.2";
+
+  buildInputs = [
+    patchelf
+    procps
+    ncurses5
+    makeWrapper
+    coreutils
+  ];
+
+  inherit ncurses5;
+
+  # requireFile prevents rehashing each time, which saves time during
+  # rebuilds.
+  src = requireFile rec {
+    name = "Xilinx_Vivado_SDK_2017.2_0616_1.tar.gz";
+    message = ''
+      This nix expression requires that ${name} is already part of the store.
+      Login to Xilinx, download from
+      https://www.xilinx.com/support/download.html,
+      rename the file to ${name}, and add it to the nix store with
+      "nix-prefetch-url file:///path/to/${name}".
+    '';
+    sha256 = "06pb4wjz76wlwhhzky9vkyi4aq6775k63c2kw3j9prqdipxqzf9j";
+  };
+
+  patchPhase = ''
+    patchShebangs .
+    patchelf --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" \
+         tps/lnx64/jre/bin/java
+  '';
+
+  dontBuild = true;
+
+  installPhase = ''
+    # generate an installation config file
+    cat <<EOF > install_config.txt
+    Edition=Vivado HL WebPACK
+    Destination=$out/opt
+    Modules=Software Development Kit (SDK):1,DocNav:0,Kintex UltraScale:0,Zynq-7000:1,System Generator for DSP:0,Artix-7:1,Kintex-7:0
+    InstallOptions=Acquire or Manage a License Key:0,Enable WebTalk for SDK to send usage statistics to Xilinx:0
+    CreateProgramGroupShortcuts=0
+    ProgramGroupFolder=Xilinx Design Tools
+    CreateShortcutsForAllUsers=0
+    CreateDesktopShortcuts=0
+    CreateFileAssociation=0
+    EOF
+
+    ($(pwd)/xsetup --agree 3rdPartyEULA,WebTalkTerms,XilinxEULA --batch Install --config install_config.txt || true) | while read line
```

```suggestion
    (./xsetup --agree 3rdPartyEULA,WebTalkTerms,XilinxEULA --batch Install --config install_config.txt || true) | while read line
```

Why do we allow it to fail?

---

I don't know, I didn't write it. But, I suppose I can test what happens when it does fail.